packages/dds/map/src/directory.ts

```diff
@@ -832,15 +841,63 @@ export class SharedDirectory extends SharedObject<ISharedDirectoryEvents> implem
     }
 }
 
+interface IKeyEditLocalOpMetadata {
+    type: "edit";
+    pendingMessageId: number;
+    // "unknown" indicates we don't know the prior value because it is a stashed op so it cannot be rolled back
+    previousValue: ILocalValue | undefined | "unknown";
```

Comment:
would null be a good choice here? it has the advantage of evaluating the same way as undefined in truthy checks, unlike the string which requires its own comparison.

i also don't really understand the scenario, but maybe it's that i don't fully understand stashed ops. in the stashed op case aren't the stashed operations locally applied to the dds before they are submitted such, in which case the correct metadata should be generated then?

Reply:
I'm trying to be super explicit about the two cases: there was no previous value of the key/subdirectory before the set/create _versus_ we have no idea what the previous value might have been. In the former case, we should remove the entry while rolling back. In the latter case, we fail to rollback. So I don't want truthy/falsy matches.

It looks like we always get local metadata for stashed ops while applying the stashed op. In that case, the DDS should be in the same state as before and we could get the data for rolling back while/before applying the edit. Although not sure if we would actually want to since rollback scenario should never happen for stashed ops even if we can technically do it. @wes-carlson 

So that leaves at least three options:

1. leave it the way it is here or other explicit indicator
2. don't have super explicit way of of indicating unrollback-able localOpMetadata (use null or undefined) and rely on that being prevented elsewhere at a higher level. If somehow rollback is called, the DDS can have wrong edit without being aware of an error (e.g. deletes a key rather than setting prior value).
3. set localOpMetadata correctly even for stashed ops [and DDS will just happily roll them back if requested], and rely on that being prevented elsewhere.

Reply:
As discussed, I removed all no-prior-localOpMetadata handling since this shouldn't happen, even with stashed ops.